### PR TITLE
Make tests compatible with venv bundle

### PR DESCRIPTION
**Pull request by m-czernek**
### What does this PR do?

In this PR, we fix the `test_pip` and `test_fileserver` tests to be compatible with venv minion bundle.

This PR is a part of https://github.com/SUSE/spacewalk/issues/23536.

NOTE: This change includes the following commit: https://github.com/openSUSE/salt/pull/662/commits/96391b8784f24fe79b576dfd94fd994502583e04

This commit will be reverted before merging. The commit modifies the skip list temporarily to actually test the modified tests.

In https://raw.githubusercontent.com/openSUSE/salt-test-skiplist/testing-failing-tests/skipped_tests.toml, I have commented out tests that are fixed by this PR for better readability. 

### Commits signed with GPG?
Yes/No

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Review by meaksh — COMMENTED** [review]
Do you think any of these changes makes sense upstream?

**Comment by m-czernek** [comment]
Possibly yes - the code changes should work with upstream too, though I'm not sure if they'll accept them. In any case, I created PRs at:

- https://github.com/saltstack/salt/pull/66703
- https://github.com/saltstack/salt/pull/66704

**Comment by m-czernek** [comment]
Note: After the PR is merged, this issue returns to backlog, since there are still more tests to take a look at.

After the PR is merged, we should wait for the next Salt minor release and then merge https://github.com/openSUSE/salt-test-skiplist/tree/testing-failing-tests to enable the tests for our Jenkins pipelines.

**Review by agraul — APPROVED** [review]
Generally LGTM! I left a few comments inline, but they aren't blocking this PR.

File: tests/pytests/functional/modules/test_sdb.py
Comment (m-czernek) [reply tests/pytests/functional/modules/test_sdb.py:19]: It disables the test. 

The test is failing, upstream is skipping the test with this mark => imho, it doesn't make sense to fix the test when upstream is OK with the test not running.

Comment (agraul) [reply tests/pytests/functional/modules/test_sdb.py:19]: I didn't realize you backported this change from upstream. Fine to keep it then.

File: tests/pytests/unit/test_fileserver.py
Comment (m-czernek) [reply tests/pytests/unit/test_fileserver.py:79]: Nice catch, updating the formatting here. 

I'll be honest that I don't totally understand why we need this. My limited understanding is that because we run the tests from a venv, it adds an additional level for the salt loader module. I didn't investigate the root cause, i.e. what exactly happens at the salt loader level when we run this from venv so I might be wrong here.

**Comment by m-czernek** [comment]
Squashing commits before merge such that:

- I remove the modifications to the workflow files.
- I squash my commits into one.
- I leave cherry-picked commits.

**Pull request merged**
